src/common/Makefile-common.am

```diff
@@ -243,7 +243,7 @@ nodist_noinst_DATA += libpreload-temp-home.so
 EXTRA_DIST += src/common/preload-temp-home.c
 
 libpreload-temp-home.so: src/common/preload-temp-home.c
-	$(AM_V_CCLD) $(CC) $(CFLAGS) -fPIC -shared -o $@ $^
+	$(AM_V_CCLD) $(CC) $(AM_CPPFLAGS) $(CFLAGS) -fPIC -shared -o $@ $^
```

Comment:
The commit message doesn't explain what this is about.

Reply:
Right, this can be dropped now; it was necessary before so that this custom .c file build rule would actually use the `_GNU_SOURCE` from `AM_CPPFLAGS`, but that's not necessary any more. I'll drop it. It also fails to build now, will investigate (not obvious why at first sight)

Reply:
We need this after all, but I moved it to a separate commit and consistently fixed it for all custom compiler invocations.